platform/persistence/platform-persistence-core-impl/src/main/java/org/codice/ddf/persistence/internal/PersistentStoreImpl.java

```diff
@@ -157,8 +165,10 @@ private void doRollback(SolrClient solrClient, String type) {
         throw new PersistenceException("Unsupported query " + cql);
       }
       QueryResponse solrResponse = solrClient.query(solrQuery, METHOD.POST);
-      long numResults = solrResponse.getResults().getNumFound();
-      LOGGER.debug("numResults = {}", numResults);
+
+      if (getAllResults) {
+        solrQuery.setRows(getNumResults(solrClient, solrQuery));
```

Comment:
❗️ What if there are a million results?  I am worried this will create a new attack path.  @codice/security 

What persistence store is missing is a way to do paging.  We could add a method with start index and page size parameters like Catalog queries.

https://github.com/codice/ddf/blob/master/catalog/core/catalog-core-api/src/main/java/ddf/catalog/operation/Query.java#L28-L49

Reply:
I changed to a page size / start index implementation.